### PR TITLE
Throw on invalid platform return type

### PR DESCRIPTION
**Pull request by krocard**
## Problem (PFL-69)
If the type returned from the platform side is not the expected one, `invokeMethod returns null`. We are handling it by returning default values.

## Changes
As this should never happen, and if it was to, it would be a bug, throw an exception instead of hiding the issue.

## Checklist
- [x] 🗒 `CHANGELOG` entry if applicable

### CODE REVIEW

File: lib/src/player.dart [hunk 1]
```diff
@@ -143,7 +154,7 @@ class Player with PlayerEventHandler implements PlayerApi {
   ]) async {
     final result = await _initializationResult;
     if (!result) {
-      return Future.error('Error initializing player on native platform side.');
```
Comment (krocard) [inline lib/src/player.dart:146]: Wrapping in `Future.error` is unnecessary in an async function, it is equivalent to `throw`.
Additionally, the code was throwing a string when Dart recommends to only throw subclasses of Exception or Error.

File: lib/src/player.dart [hunk 2]
```diff
@@ -123,17 +123,28 @@ class Player with PlayerEventHandler implements PlayerApi {
...
     return _methodChannel.invokeMethod<T>(methodName, _buildPayload(data));
   }
 
+  Future<T> _invokeMethod<T>(
+    String methodName, [
+    dynamic data,
+  ]) async {
+    final result = await _invokeMethodNullable<T>(methodName, data);
+    if (result == null) {
+      throw Exception('Invalid type returned from the native platform side.');
```
Comment (hawk23) [inline lib/src/player.dart:143]: Let's also print the method name for better debugging:
```suggestion
      throw Exception('Invalid type returned for $methodName from the native platform side.');
```

**Review by hawk23 — COMMENTED** [review]
Nice improvement! 💪 

During manual testing, I found a lot of those "Invalid type" exceptions in the console. A quick check revealed that now all calls of `_invokeMethod<void>()` fail with an exception now. I tested this on iOS but it should also fail on Android. 

Also, `flutter test integration_test` fails now (we really should execute those in the CI soon 😅).

Some failing examples from `player.dart`
```
@override
Future<void> play() async => _invokeMethod<void>(Methods.play);

@override
Future<void> mute() async => _invokeMethod<void>(Methods.mute);

@override
Future<void> unmute() async => _invokeMethod<void>(Methods.unmute);

@override
Future<void> pause() async => _invokeMethod<void>(Methods.pause);
```

File: lib/src/player.dart (hunk 2 above)
Comment (krocard) [reply lib/src/player.dart]: Done

**Comment by krocard** [comment]
> During manual testing, I found a lot of those "Invalid type" exceptions in the console. A quick check revealed that now all calls of _invokeMethod<void>() fail with an exception now. I tested this on iOS but it should also fail on Android.

Good catch. I was expecting the app to crash if any error was raised. I did not realized they would be logged. The issue was due to void being the only type for which `T == T?`. The problem should be fixed now.

> Also, flutter test integration_test fails now (we really should execute those in the CI soon 😅).

I have not been able to run this the test locally. I guess some weird error:
~~~ bash
$ flutter test integration_test
00:00 +0: loading /home/krocard/dev/bitmovin/bitmovin-player-flutter/example/integration_test/seeking_test.dart                                                                                                                        R../player_testing/lib/env/env.dart:3:6: Error: Error when reading '../player_testing/lib/env/env.g.dart': No such file or directory
part 'env.g.dart';
     ^
../player_testing/lib/env/env.dart:3:6: Error: Can't use '../player_testing/lib/env/env.g.dart' as a part, because it has no 'part of' declaration.
part 'env.g.dart';
     ^
../player_testing/lib/env/env.dart:14:50: Error: Undefined name '_Env'.
  static const String bitmovinPlayerLicenseKey = _Env.bitmovinPlayerLicenseKey;
~~~
But it does have a `part of`:
~~~
$ head -n3 ../player_testing/lib/env/env.dart
import 'package:envied/envied.dart';

part 'env.g.dart';
~~~

**Comment by hawk23** [comment]
> I have not been able to run this the test locally. I guess some weird error:

Please have look at `Integration Tests` section in `CONTRIBUTING.md`. You need to create the `.env` file and run `dart run build_runner build` once to solve this.

**Review by hawk23 — APPROVED** [review]

File: lib/src/player.dart (hunk 2 above)
Comment (krocard) [reply lib/src/player.dart]: Done

**Pull request merged**